components/pkg-export-docker/src/os/check/windows.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{error::Result,
+            util};
+
+/// Currently when exporting containers on Windows, the Docker daemon
+/// *must* be in Windows mode (i.e., only Windows containers can be
+/// exported on Windows machines).
+///
+/// If the daemon is in Linux mode, we return an error and should stop
+/// the export process.
+pub(crate) fn ensure_proper_docker_platform() -> Result<()> {
+    match DockerOS::current() {
+        DockerOS::Windows => Ok(()),
+        other => Err(Error::DockerNotInWindowsMode(other).into()),
+    }
+}
+
+#[derive(Debug, Fail)]
+enum Error {
+    #[fail(display = "Only Windows container export is supported; please set your Docker daemon \
```

Comment:
Very excited to standardize our error handling! Rust's error handling has undergone a lot of churn. I think currently my preferred approach is to use [`thiserror`](https://github.com/dtolnay/thiserror) in library code and [`anyhow`](https://github.com/dtolnay/anyhow) in application code. For my motivation on when to use each see [here](https://github.com/dtolnay/thiserror#comparison-to-anyhow). For my motivation for not using `failure` see [here](https://github.com/dtolnay/anyhow#comparison-to-failure). What are your thoughts?

Reply:
I haven't seen `thiserror` or `anyhow` yet. I am aware that `failure` seems to not the the way to go, though, and would love to move this particular crate away from it.

I'll check these out!